### DISCUSSION TITLE
Does leaf useApache Log4j 2?

### DISCUSSION THREAD

**Discussion by glabrie10** [uwrit/leaf#485] [Q&A] [answered]
Recently Apache Log4j 2 has been flagged as a vulnerability.  Is this package in use by the leaf application?

**Comment by ndobb** [comment, marked as answer]
Hi @glabrie10, no, the Leaf API is written in C# and does not use Log4j (which to my knowledge is only used in Java). For reference, the logging library used in Leaf is [Serilog](https://github.com/serilog/serilog).
